local-cluster/tests/local_cluster.rs

```diff
@@ -1677,7 +1678,11 @@ fn do_test_optimistic_confirmation_violation_with_or_without_tower(with_tower: b
     let node_stakes = vec![31, 36, 33, 0];
 
     // Each pubkeys are prefixed with A, B, C and D.
-    // D is needed to avoid NoPropagatedConfirmation erorrs
+    // D is needed to:
+    // 1) Propagate A's votes for S2 to validator C after A shuts down so that
+    // C can avoid NoPropagatedConfirmation erorrs and continue to generate blocks
+    // 2) Provide gosisp discovery for `A` when it restarts because `A` will restart
+    // at a different gossip port than the entrypoint saved in C's gosisp table
```

Comment:
```suggestion
    // C can avoid NoPropagatedConfirmation errors and continue to generate blocks
    // 2) Provide gossip discovery for `A` when it restarts because `A` will restart
    // at a different gossip port than the entrypoint saved in C's gossip table
```

You still blaming the keyboard? :laughing:

Reply:
Oh, that's just how I thought those were words were spelled, totally on purpose